libs/features/richresults.js

```diff
@@ -1,26 +1,60 @@
-// createTag and getMetadata are passed in to avoid circular dependencies
-export function addRichResults(type, { createTag, getMetadata }) {
-  const addRichResultsForNewsArticle = () => {
-    const newsArticle = {
-      '@context': 'https://schema.org',
-      '@type': 'NewsArticle',
-      headLine: getMetadata('og:title'),
-      image: getMetadata('og:image'),
-      datePublished: getMetadata('published'),
-      dateModified: getMetadata('modified'),
-      author: {
-        '@type': 'Person',
-        name: getMetadata('authorname'),
-        url: getMetadata('authorurl'),
+function getRichResultsForNewsArticle(getMetadata) {
+  return {
+    '@context': 'https://schema.org',
+    '@type': 'NewsArticle',
+    headLine: getMetadata('og:title'),
+    image: getMetadata('og:image'),
+    datePublished: getMetadata('published'),
+    dateModified: getMetadata('modified'),
+    author: {
+      '@type': 'Person',
+      name: getMetadata('authorname'),
+      url: getMetadata('authorurl'),
+    },
+  };
+}
+
+function getRichResultsForSiteSearchBox(getMetadata) {
+  // See specifications at https://developers.google.com/search/docs/appearance/structured-data/sitelinks-searchbox
+  const SEARCH_TERM_STRING = 'search_term_string';
+  const urlTemplate = `${getMetadata('search-url')}?${getMetadata('search-parameter-name')}={${SEARCH_TERM_STRING}}`;
+  return {
+    '@context': 'https://schema.org',
+    '@type': 'WebSite',
+    url: getMetadata('url'),
+    potentialAction: {
+      '@type': 'SearchAction',
+      target: {
+        '@type': 'EntryPoint',
+        urlTemplate: urlTemplate,
       },
-    };
-    const script = createTag('script', { type: 'application/ld+json' }, JSON.stringify(newsArticle));
-    document.head.append(script);
-  }
-  
+      'query-input': `required name=${SEARCH_TERM_STRING}`,
+    },
+  };
+}
+
+function getRichResults(type, getMetadata) {
   switch (type) {
-    // add support for new types here, e.g. Product
     case 'NewsArticle':
-      addRichResultsForNewsArticle();
+      return getRichResultsForNewsArticle(getMetadata);
+    case 'SiteSearchBox':
+      return getRichResultsForSiteSearchBox(getMetadata);
+    default:
+      // eslint-disable-next-line no-console
+      console.error(`Type ${type} is not supported`);
+      return null;
   }
 }
+
+function addToDom(richResults, createTag) {
+  if (richResults) {
+    const script = createTag('script', { type: 'application/ld+json' }, JSON.stringify(richResults));
+    document.head.append(script);
+  }
```

Comment:
nit: in general we try to use early return to avoid nesting
```suggestion
  if (!richResults) return;
  
  const script = createTag('script', { type: 'application/ld+json' }, JSON.stringify(richResults));
  document.head.append(script);
```

Reply:
thx, addressed in https://github.com/adobecom/milo/pull/389/commits/6cec504e9e1558b4f77f96ccf947ad8acfd94c77